Python/Product/PythonTools/PythonTools/Intellisense/BufferParser.cs

```diff
@@ -303,31 +294,37 @@ sealed class BufferParser : IDisposable {
 
         internal static async Task ParseBuffersAsync(
             PythonEditorServices services,
-            AnalysisEntry entry,
+            VsProjectAnalyzer analyzer,
             IEnumerable<ITextSnapshot> snapshots
         ) {
-            var updates = snapshots.Select(s => GetUpdateForSnapshot(services, s)).Where(u => u != null).ToList();
+            var updates = snapshots
+                .GroupBy(s => PythonTextBufferInfo.TryGetForBuffer(s.TextBuffer)?.AnalysisEntry.FileId ?? -1)
+                .Where(g => g.Key >= 0)
+                .Select(g => Tuple.Create(g.Key, g.Select(s => GetUpdateForSnapshot(services, s)).ToArray()))
+                .Where(u => u != null).ToList();
```

Comment:
>Where(u => u != null) [](start = 17, length = 21)

I'm confused as to what 'u' is now. It used to be the update object. Is that still the case?

Reply:
Great catch! This code is somewhat broken now (though I guess we can handle null updates). I'll fix

Reply:
I mean it looks like it's the tuple now, which would never be null?

---
In reply to: [135308418](https://github.com/Microsoft/PTVS/pull/2936#discussion_r135308418) [](ancestors = 135308418)